### PR TITLE
OCPCLOUD-2565: Updates features to support adding to FlagSet

### PR DESCRIPTION
**Pull request by theobarberbany**
This updates `pkg/features` to support adding flags to a flagset, rather than just a cobra flag.

This is useful because in some projects we don't use cobra, e.g MAO and it's operands.

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -39,7 +40,16 @@ func NewFeatureGateOptionsOrDie(featureGates featuregate.MutableFeatureGate, pro
...
 	flags := cmd.Flags()
-	flags.Var(flag.NewMapStringBool(&o.featureGateArgs), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+	flags.Var(k8sflag.NewMapStringBool(&o.featureGateArgs), "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
+		"Options are:\n"+strings.Join(o.featureGates.KnownFeatures(), "\n"))
+}
+
+func (o *FeatureGateOptions) AddFlagsToGoFlagSet(flagset *flag.FlagSet) {
+	if flagset == nil {
+		flagset = flag.CommandLine
+	}
```
Comment (bertinatto) [inline pkg/features/features.go:50]: Are you sure this is going to work? 

This is just changing the local copy of `flagset`, the original variable will continue to be nil.

**Review by bertinatto — COMMENTED** [review]
@theobarberbany please take a look at my comment before merging.

**Comment by bertinatto** [comment]
/hold
for https://github.com/openshift/library-go/pull/1775#discussion_r1722172893

File: pkg/features/features.go (hunk 1 above)
Comment (JoelSpeed) [reply pkg/features/features.go:50]: `flag.CommandLine` is a global variable, so by assigning here, we are effectively calling `flag.CommandLine.Var` on line 52, which is the intended use to add global flags, so I'm pretty sure this is ok

Comment (theobarberbany) [reply pkg/features/features.go:50]: Ah yes, I thought the same at first as well! It's not clear that `flag.CommandLine` is a global var :) It's the same thing that `klog` does when you do a `klog.InitFlags(nil)`

I've also verified it works in a local build.

Comment (bertinatto) [reply pkg/features/features.go:50]: Ack, that makes sense. Sorry for the noise.
/hold cancel

Comment (theobarberbany) [reply pkg/features/features.go:50]: No worries! Do you know who I should ping for approval? :)

Comment (bertinatto) [reply pkg/features/features.go:50]: CC @soltysh

**Comment by JoelSpeed** [comment]
@bertinatto Given your comment was resolved, are you able to approve?

**Comment by bertinatto** [comment]
> @bertinatto Given your comment was resolved, are you able to approve?

Trying for the first time here...

/approve

**Pull request merged**